quic/s2n-quic-core/src/path/mtu.rs

```diff
@@ -259,6 +312,36 @@ impl Config {
     }
 }
 
+// A checked MTU Config created after validating the values from the IO provider
+// against the values from mtu provider.
+#[derive(Copy, Clone, Debug, Default)]
+pub struct CheckedConfig(Config);
```

Comment:
do we need a new type for this? Seems like it could just be another method on `mtu:Config` that validates the config against the given endpoint `mtu::Config`

Reply:
Replied in the previous comment: https://github.com/aws/s2n-quic/pull/2232#discussion_r1638736490

Summary
- its good to validate that we are using the correct type
- since mtu::Config is now part of the public API, the new type helps to avoid exposing internal details